misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -595,20 +626,39 @@ public ImportPrefixNode transform(ImportPrefixNode importPrefixNode) {
     @Override
     public ServiceDeclarationNode transform(ServiceDeclarationNode serviceDeclarationNode) {
         MetadataNode metadata = formatNode(serviceDeclarationNode.metadata().orElse(null), 0, 1);
+        NodeList<Token> qualifiers = formatNodeList(serviceDeclarationNode.qualifiers(), 1, 0, 1, 0);
         Token serviceKeyword = formatToken(serviceDeclarationNode.serviceKeyword(), 1, 0);
-        //IdentifierToken serviceName = formatToken(serviceDeclarationNode.serviceName().orElse(null), 1, 0);
+        TypeDescriptorNode typeDescriptor = formatNode(serviceDeclarationNode.typeDescriptor().orElse(null), 1, 0);
+        NodeList<Node> absoluteResourcePath = formatNodeList(serviceDeclarationNode.absoluteResourcePath(), 0, 0, 1, 0);
         Token onKeyword = formatToken(serviceDeclarationNode.onKeyword(), 1, 0);
         SeparatedNodeList<ExpressionNode> expressions =
                 formatSeparatedNodeList(serviceDeclarationNode.expressions(), 0, 0, 1, 0);
-        //Node serviceBody = formatNode(serviceDeclarationNode.serviceBody(), env.trailingWS, env.trailingNL);
+        Token openBrace = formatToken(serviceDeclarationNode.openBraceToken(), 0, 1);
+        indent(); // increase the indentation of the following statements.
+        NodeList<Node> members = formatNodeList(serviceDeclarationNode.members(), 0, 1, 0, 1, true);
+        unindent(); // reset the indentation.
+        env.preserveNewlines = true;
+        Token closeBrace = formatToken(serviceDeclarationNode.closeBraceToken(), env.trailingWS, env.trailingNL);
 
         return serviceDeclarationNode.modify()
                 .withMetadata(metadata)
+                .withQualifiers(qualifiers)
                 .withServiceKeyword(serviceKeyword)
-                //.withServiceName(serviceName)
+                .withTypeDescriptor(typeDescriptor)
+                .withAbsoluteResourcePath(absoluteResourcePath)
                 .withOnKeyword(onKeyword)
                 .withExpressions(expressions)
-                //.withServiceBody(serviceBody)
+                .withOpenBraceToken(openBrace)
+                .withMembers(members)
+                .withCloseBraceToken(closeBrace)
+                .apply();
+    }
+
+    @Override
+    public RequiredExpressionNode transform(RequiredExpressionNode requiredExpressionNode) {
+        Token questionMarkToken = formatToken(requiredExpressionNode.questionMarkToken(), 0, 0);
```

Comment:
```suggestion
        Token questionMarkToken = formatToken(requiredExpressionNode.questionMarkToken(), env.trailingWS, env.trailingNL);
```

Reply:
@IrushiL Thanks for pointing this out. :+1: 
Fixed.